source/Calamari/Kubernetes/SetupKubectlAuthentication.cs

```diff
@@ -45,17 +48,27 @@ public CommandResult Execute(string accountType)
         {
             var errorResult = new CommandResult(string.Empty, 1);
 
-            foreach (var proxyVariable in ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables())
+            foreach (var proxyVariable in GenerateProxyEnvironmentVariables())
             {
                 environmentVars[proxyVariable.Key] = proxyVariable.Value;
             }
 
-            var kubeConfig = CreateKubectlConfig();
+            // Make sure Kubectl exists and can be run
             if (!kubectl.TrySetKubectl())
             {
                 return errorResult;
             }
 
+            // If we are running on a Kubernetes Tentacle Target,
+            // then we can rely on using the service account inside the pod for authentication.
+            if (variables.Get(MachineVariables.DeploymentTargetType) == KubernetesTentacleTargetTypeId)
```

Comment:
Im assuming that this variable `MachineVariables.DeploymentTargetType` is different if the target is being used as a worker? 
Wondering what this would mean in future if we were running workloads on a KubernetesTarget worker but trying to connect to an external cluster? 
At the moment we will just ignore any of those other credentials used before. Is that what we want?

Reply:
In the original design of the PR we had a variable that was going to be contributed to by the deployment target.

@scme0 Do you know the answer to robs question?

Reply:
If you are executing Kubernetes steps on a Kubernetes Tentacle as a worker you would be using a Kubernetes API target so the Deployment Target Type would be `"Kubernetes"` I believe.

To be honest we could make it stricter, we could say "Target Type" == "Kubernetes" do auth stuff, else if "KubernetesTentacle" skip and log, else throw exception "Unsupported Target for Kubernetes Authentication".

Reply:
I guess people could be just adding their own variables in to get this auth to work but they really should be using the build in Kubernetes API target in that case.

Reply:
@APErebus @zentron So after thinking about this more I'm still of the opinion that this is ok. The only way we allow users to authenticate against a kubernetes cluster is by using a kubernetes target (old or new). For new targets, they will not be executing on a worker because it's a tentacle and for the old one, it doesn't matter where it executes as it will be executing on behalf of the old target and this property will be set to "Kubernetes" and so it will run through the authentication as expected.

Reply:
You could be right in terms of "it should just work", but we don't typically do machine specific behavioral logic within calamari and this could be unexpected, particularly when this approach still allows users to send credentials etc but it just gets ignored. 

My first thought was why not just work through the config flow and basically not create the file if no credentials are provided (which they shouldn't be in this case) so that the data itself drives the logic rather than a case of it just happening to work due to the nuances of how workers and targets are used.

In any case since this is a two way door decision, once you have had a quick investigation of a more data-driven approach suggested, I'm also fine for this comment to not be a blocker in moving forward for the time being.

My only additional comment was going to be lets make sure we have a log entry there but I can see you have that base covered already :)